fe/fe-core/src/main/java/com/starrocks/persist/Storage.java

```diff
@@ -117,30 +104,29 @@ public void reload() throws IOException {
         }
 
         // Find the latest image
-        File dir = new File(metaDir);
+        imageSeq = getImageSeq(metaDir);
+    }
+
+    public static long getImageSeq(String imageDir) {
+        long imageSeq = 0;
+        File dir = new File(imageDir);
         File[] children = dir.listFiles();
         if (children == null) {
-            return;
+            return imageSeq;
         } else {
             for (File child : children) {
                 String name = child.getName();
                 try {
-                    if (!name.equals(EDITS) && !name.equals(IMAGE_NEW)
-                            && !name.endsWith(".part") && name.contains(".")) {
-                        if (name.startsWith(IMAGE)) {
-                            imageSeq = Math.max(Long.parseLong(name.substring(name.lastIndexOf('.') + 1)), imageSeq);
-                        } else if (name.startsWith(EDITS)) {
-                            // Just record the sequence part of the file name
-                            editsFileSequenceNumbers.add(Long.parseLong(name.substring(name.lastIndexOf('.') + 1)));
-                            editsSeq = Math.max(Long.parseLong(name.substring(name.lastIndexOf('.') + 1)), editsSeq);
-                        }
+                    if (name.startsWith(IMAGE) && name.contains(".")) {
```

Comment:
why to read a file but not store it in memory after checkpoint

Reply:
I have ever thought this method before. But is a little complex: cause checkpoint is done by master, and we should update the memory when master accomplish the checkpoint, non-master got the image push, and all-node start-up. Metrics api is not a high-load api, so get it from disk may be an easy and robust method.

Reply:
But it's will have high latency to read file.
And also it will read temporary file during checkpoint operation.